doc/contributing/docs/api.rst

```diff
@@ -0,0 +1,170 @@
+
+Style
+-----
+
+Please write as simply as possible. Describe functionality using short sentences in the present simple tense.
+A short sentence consists of no more than two clauses.
+Consider using `LanguageTool <https://languagetool.org/>`_ or `Grammarly <https://www.grammarly.com/>`_
+to check your English.
+For more style-related specifics, consult the :doc:`Language and style </contributing/docs/style>` section.
+
+..  _contributing-docs-api_version:
+
+Indicating the version
+----------------------
+
+For every new feature or parameter, please indicate the version it was introduced in.
+
+With Tarantool features and parameters, use one of the following Sphinx directives:
+
+..  code-block:: rst
+
+    Since :tarantool-release:`2.10.0`.
+    This is a link to the release notes on GitHub.
+
+    Since :doc:`2.10.0 </release/2.10.0>`.
+    This is a link to the release notes on the Tarantool documentation website.
+
+The result looks like this:
+
+    Since Tarantool :tarantool-release:`2.10.0`.
+    This is a link to the release notes on GitHub.
+
+    Since Tarantool :doc:`2.10.0 </release/2.10.0>`.
+    This is a link to the release notes on the Tarantool documentation website.
+
+See also the :doc:`guide on writing release notes </contributing/release_notes/>`.
+
+..  _contributing-api-docs_general-description:
+
+Language of the general description
+-----------------------------------
+
+Use one of the two options:
+
+*   Start with a verb in the imperative mood. Example: *Create a fiber.*
+*   Start with a noun. Example: *The directory where memtx stores snapshot files.*
+
+Checklist
+---------
+
+Function or method
+^^^^^^^^^^^^^^^^^^
+
+*   :ref:`Since which Tarantool version <contributing-docs-api_version>`
+*   :ref:`General description <contributing-api-docs_general-description>`
+*   :ref:`Parameters <documenting_parameters>`
+*   What this function returns (if nothing, write 'none')
+*   Return type (if exists)
+*   Possible errors (if exist)
+*   Complexity factors (if exist)
+*   Usage with memtx and vinyl (if differs)
+*   Example(s)
+*   Extra information (if needed)
+
+See :ref:`module function example <contributing-api-docs_function-example>`,
+:ref:`class method example <contributing-api-docs_class-example>`.
+
+Data
+^^^^
+
+*   :ref:`Since which Tarantool version <contributing-docs-api_version>`
+*   :ref:`General description <contributing-api-docs_general-description>`
+*   Return type
+*   Example
+
+See :ref:`class data example <contributing-api-docs_class-example>`.
+
+..  _documenting_parameters:
+
+Function and class parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+*   :ref:`Since which Tarantool version <contributing-docs-api_version>`
+    (if added later)
+*   :ref:`General description <contributing-api-docs_general-description>`
+*   Type
+*   Default value (if optional), possible values
+
+If the parameter is optional, make sure it is enclosed in square brackets
+in the function declaration (in the "heading").
```

Comment:
That's good this guideline is in general description for all the parameters now!

Maybe we can give an example here? Because it's not clear what "heading" means in this case